pkg/routeagent/controllers/route/route.go

```diff
@@ -318,14 +319,17 @@ func (r *Controller) createVxLANInterface(ifaceType int, gatewayNodeIP net.IP) e
 		return fmt.Errorf("failed to derive the vxlan vtepIP for %s, %v", ipAddr, err)
 	}
 
+	// Derive the MTU based on the default outgoing interface
+	vxlanMtu := r.defaultHostIface.MTU - VxLANOverhead
```

Comment:
I'm trying to think if in some environments this would not be perfect, for example, deployments where it some hosts will span multiple L2 domains, where some of the links are smaller.

There's no simple solution for such thing.  May be we can add an environment variable to mitigate that, and let people configure down the MTU size?

May be we can add a log entry explaining from which interface we derived the MTU, and the mtu assigned to vxlan?


Variables can be added here: https://github.com/submariner-io/submariner/blob/929d9bf1991bb67b7dcc2e351644a6585d7ad69c/pkg/types/types.go#L16

And captured here:
https://github.com/submariner-io/submariner/blob/929d9bf1991bb67b7dcc2e351644a6585d7ad69c/main.go#L55

like SUBMARINER_NAT_ENABLED, etc...

May be I'm going to far, but I'm trying to think of past issues we encoutered with neutron/etc, that then we make easier to mitigate in the field if happen.

Reply:
@mangelajo I see your point, but I'm not sure if such environments are commonly used in cloud environments. My main concern is about introducing **new environment variables** which will affect the user-experience. I agree that we can/should do it if necessary, but I feel that we can hold on until we come across such a use-case - at the most the effect would be slightly degraded performance (submariner functionality would still continue as you know).

Reply:
Fair point. Give that the worst scenario is likely to be slight performance degradation in those corner cases I agree with you that making UX worse it's better to wait and see if that's really ever a problem.